comps/dataprep/redis/langchain/prepare_doc_redis.py

```diff
@@ -187,5 +211,21 @@ def process_files_wrapper(files):
     raise HTTPException(status_code=400, detail="Must provide either a file or a string list.")
 
 
+@register_microservice(
+    name="opea_service@prepare_doc_redis_file", endpoint="/v1/dataprep/get_file", host="0.0.0.0", port=6008
+)
+@traceable(run_type="tool")
+async def rag_get_file_structure():
+    print("[ get_file_structure] ")
+
+    if not Path(upload_folder).exists():
+        print("No file uploaded, return empty list.")
+        return []
+
+    file_content = get_file_structure(upload_folder)
```

Comment:
This will return all the files uploaded even if embedding are not creating for them in Redis. is this intended?

Reply:
What is the purpose of this API? It is to return ingested files or files to be ingested?

Reply:
@letonghan: What happens if some of the file ingestion to Redis fails for some reason?

Reply:
@letonghan I notice couple of the review comments are marked resolved without any response. Can you pls add response?